src/titiler/core/titiler/core/templates/index.html

```diff
@@ -69,79 +36,63 @@
     }
 }
 
-const addAOI = (bounds) => {
-  if (map.getLayer('aoi-polygon')) map.removeLayer('aoi-polygon')
-  if (map.getSource('aoi')) map.removeSource('aoi')
-  const geojson = {
-      "type": "FeatureCollection",
-      "features": [bboxPolygon(bounds)]
+// tms comes from morecantile
+// https://github.com/developmentseed/morecantile/blob/main/morecantile/models.py
+var crs = new L.Proj.CRS(
+    '{{ tms.crs.srs }}',
+    '{{ tms.crs.to_proj4() }}', {
+    origin: [{{ tms.xy_bbox[0] }}, {{ tms.xy_bbox[3] }}],
+    // bounds: [[{{ tms.xy_bbox[0] }}, {{ tms.xy_bbox[1] }}], [{{ tms.xy_bbox[2] }}, {{ tms.xy_bbox[3] }}]],
+    resolutions: {{ resolutions|safe }},
```

Comment:
⚠️ need help here ⚠️ 

It seems to work but it needs more testing

Reply:
I'm not sure if `resolutions` is https://github.com/developmentseed/morecantile/blob/main/morecantile/models.py#L370-L379

Reply:
So this is a bit tricky.  I'll quote myself from https://gis.stackexchange.com/a/423647/53755 :

> The result of `L.CRS.scale()` has to be multiplied by a CRS-dependant constant (the world bounds according to that CRS) in order to make the numbers meaningful, which IMO leads to confusion.

So you have to be careful about the CRS's bounds, specially if you plan on using TMS-style tiles with an inverted Y-coordinate (a `{-y}` in the URL template). Specifically, see https://stackoverflow.com/a/62625435/4768502

See also https://gis.stackexchange.com/a/310124/53755 - Don't be confused when using Leaflet's functions for projecting/unprojecting coordinates because they don't rely on CRS-relative numbers, but rather screen-relative numbers.

Reply:
Thanks for you help @IvanSanchez 
going back at this today, and I'm not quite sure I understand all of ☝️ 😭